pkg/core/blockchain.go

```diff
@@ -1241,11 +1241,97 @@ func (bc *Blockchain) verifyTx(t *transaction.Transaction, block *block.Block) e
 		if bc.dao.IsDoubleClaim(claim) {
 			return errors.New("double claim")
 		}
+		if err := bc.verifyClaims(t); err != nil {
+			bc.log.Warn("error while verifying claims", zap.String("error", err.Error()))
+		}
 	}
 
 	return bc.verifyTxWitnesses(t, block)
 }
 
+func (bc *Blockchain) verifyClaims(tx *transaction.Transaction) (err error) {
+	t := tx.Data.(*transaction.ClaimTX)
+	var result *transaction.Result
+	results := bc.GetTransactionResults(tx)
+	for i := range results {
+		if results[i].AssetID == UtilityTokenID() {
+			result = results[i]
```

Comment:
Hmm, what if there is more than one GAS output?

Reply:
We will ignore the rest, probably.
https://github.com/neo-project/neo/blob/master-2.x/neo/Network/P2P/Payloads/ClaimTransaction.cs#L67

Reply:
We will ignore the rest.
https://github.com/neo-project/neo/blob/master-2.x/neo/Network/P2P/Payloads/ClaimTransaction.cs#L67

Reply:
We will ignore the rest.
https://github.com/neo-project/neo/blob/master-2.x/neo/Network/P2P/Payloads/ClaimTransaction.cs#L67